### PR TITLE
Add skin settings to database

### LINKED ISSUE #4922: Add SkinInfo key to databased Settings table
Skin settings need to be stored at the database. As a first step, they should still be re-parsed every time from the `skin.ini` as we have not yet got support for all settings to be migrated across. Eventually they will be read once from the ini and then the file will be discarded.

### PR DESCRIPTION
**Pull request by peppy**
Closes #4922.

### CODE REVIEW

File: osu.Game/Migrations/20190525060824_SkinSettings.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+                name: "SkinInfoID",
+                table: "Settings",
+                nullable: true);
+
+            migrationBuilder.CreateIndex(
+                name: "IX_Settings_SkinInfoID",
+                table: "Settings",
+                column: "SkinInfoID");
+
+            // unsupported by sqlite
```
Comment (smoogipoo) [inline osu.Game/Migrations/20190525060824_SkinSettings.cs]: So why is this here then? How are we using foreign keys everywhere else just fine but this one doesn't work? Why did the migration auto-generate this one here and not in all the other cases?

File: osu.Game/Migrations/20190525060824_SkinSettings.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+            //     table: "Settings",
+            //     column: "SkinInfoID",
+            //     principalTable: "SkinInfo",
+            //     principalColumn: "ID",
+            //     onDelete: ReferentialAction.Restrict);
+        }
+
+        protected override void Down(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.DropForeignKey(
```
Comment (smoogipoo) [inline osu.Game/Migrations/20190525060824_SkinSettings.cs]: Dropping a foreign key that wasn't added?

**Review by smoogipoo — CHANGES_REQUESTED** [review]

File: osu.Game/Migrations/20190525060824_SkinSettings.cs (hunk 1 above)
Comment (peppy) [reply osu.Game/Migrations/20190525060824_SkinSettings.cs]: as discussed in real life (all previous foreign key additions were performed along with a `CreateTable`).

i'll see if i can do a manual migration this time.

**Comment by peppy** [comment]
Switched to applying a manual migration (recommended method; sql is taken from DataGrip automatic generation as it seems most sane).

Please test against your database @smoogipoo.

**Comment by smoogipoo** [comment]
Seems to have worked fine.

**Review by smoogipoo — APPROVED** [review]
